src/pages/Token/index.tsx

```diff
@@ -69,38 +71,77 @@ const TokenPage = () => {
     fetchToken();
   }, [push, fetchOffer, fetchToken, selectedAccount?.address, collectionId, id, token]);
 
+  const backClickHandler = useCallback((event: React.MouseEvent<HTMLAnchorElement>) => {
+    // if user opened token page with a direct link, redirect him to /market, otherwise redirect him back
+    if (history.length > 2) {
+      history.back();
+      event.preventDefault();
+    }
+  }, []);
+
   if (!isFetchingOffer && !isFetchingToken && !token && !offer) return <Error404 />;
 
   // TODO: split into more categories here instead of just "buy/sell" and putting splitting inside them
 
-  return (<PagePaper>
-    <CommonTokenDetail
-      token={token}
-      offer={offer}
-      isLoading={isFetchingOffer || isFetchingToken}
-    >
-      <TokenTrading
-        token={token}
-        offer={offer}
-        onSellClick={onActionClick(MarketType.sellFix)}
-        onBuyClick={onActionClick(MarketType.purchase)}
-        onTransferClick={onActionClick(MarketType.transfer)}
-        onDelistClick={onActionClick(MarketType.delist)}
-        onDelistAuctionClick={onActionClick(MarketType.delistAuction)}
-        onPlaceABidClick={onActionClick(MarketType.bid)}
-        onWithdrawClick={onActionClick(MarketType.withdrawBid)}
-        onAuctionClose={onAuctionClose}
-      />
-      <TokenPageModal
-        token={token}
-        offer={offer}
-        marketType={marketType}
-        onFinish={onFinish}
-        onClose={onClose}
-      />
-    </CommonTokenDetail>
-  </PagePaper>
+  return (
+    <>
+      <StyledLink href='/market' onClick={backClickHandler}>
+        <Icon name='arrow-left' color={BlueGrey500} size={16}></Icon>
+        <p>back</p>
+      </StyledLink>
+      <TokenPagePaper>
+        <CommonTokenDetail
+          token={token}
+          offer={offer}
+          isLoading={isFetchingOffer || isFetchingToken}
+        >
+          <TokenTrading
+            token={token}
+            offer={offer}
+            onSellClick={onActionClick(MarketType.sellFix)}
+            onBuyClick={onActionClick(MarketType.purchase)}
+            onTransferClick={onActionClick(MarketType.transfer)}
+            onDelistClick={onActionClick(MarketType.delist)}
+            onDelistAuctionClick={onActionClick(MarketType.delistAuction)}
+            onPlaceABidClick={onActionClick(MarketType.bid)}
+            onWithdrawClick={onActionClick(MarketType.withdrawBid)}
+            onAuctionClose={onAuctionClose}
+          />
+          <TokenPageModal
+            token={token}
+            offer={offer}
+            marketType={marketType}
+            onFinish={onFinish}
+            onClose={onClose}
+          />
+        </CommonTokenDetail>
+      </TokenPagePaper>
+    </>
   );
 };
 
+const TokenPagePaper = styled(PagePaper)`
+  && {
+    margin-top: calc(var(--gap) / 2);
+  }
+`;
+
+const StyledLink = styled.a`
```

Comment:
make it text element instead. Since we are adding logic into it - we no longer can treat it as a link